src/Engine/Identities/IdentityModel.php

```diff
@@ -0,0 +1,212 @@
+        return $this->django_id;
+    }
+
+    /**
+     * Build with Django ID.
+     * @param int $djangoId
+     * @return IdentityModel
+     */
+    public function withDjangoId(int $djangoId): self
+    {
+        return $this->with('django_id', $djangoId);
+    }
+
+    /**
+     * Get the identity UUID.
+     * @return string
+     */
+    public function getIdentityUuid(): string
+    {
+        return $this->identity_uuid;
+    }
+
+    /**
+     * Build with identity UUID.
+     * @param string $identityUuid
+     * @return IdentityModel
+     */
+    public function withIdentityUuid(string $identityUuid): self
+    {
+        return $this->with('identity_uuid', $identityUuid);
+    }
+
+    /**
+     * Get the identity Traits.
+     * @return IdentityTraitList
+     */
+    public function getIdentityTraits(): IdentityTraitList
+    {
+        return $this->identity_traits;
+    }
+
+    /**
+     * Build with identity Traits.
+     * @param IdentityTraitList $identityTraits
+     * @return IdentityModel
+     */
+    public function withIdentityTraits(IdentityTraitList $identityTraits): self
+    {
+        return $this->with('identity_traits', $identityTraits);
+    }
+
+    /**
+     * Get the identity Features.
+     * @return IdentityFeaturesList
+     */
+    public function getIdentityFeatures(): IdentityFeaturesList
+    {
+        return $this->identity_features;
+    }
+
+    /**
+     * Build with identity Features.
+     * @param IdentityFeaturesList $identityFeatures
+     * @return IdentityModel
+     */
+    public function withidentityFeatures(IdentityFeaturesList $identityFeatures): self
+    {
+        return $this->with('identity_features', $identityFeatures);
+    }
+
+    /**
+     * Get the created date.
+     * @return DateTime
+     */
+    public function getCreatedDate(): DateTime
+    {
+        return $this->created_date;
+    }
+
+    /**
+     * Build with created date.
+     * @param DateTime $createdDate
+     * @return IdentityModel
+     */
+    public function withCreatedDate(DateTime $createdDate): self
+    {
+        return $this->with('created_date', $createdDate);
+    }
+
+    /**
+     * Get the composite key.
+     * @return string
+     */
+    public function compositeKey(): string
+    {
+        return self::generateCompositeKey($this->environment_api_key, $this->identifier);
+    }
+
+    /**
+     * Generate the composite key.
+     * @param string $envKey
+     * @param string $identifier
+     * @return string
+     */
+    public static function generateCompositeKey(string $envKey, string $identifier): string
+    {
+        return "{$envKey}_{$identifier}";
+    }
+
+    /**
+     * Replace/remove the identity traits.
+     * @param array $traits
+     * @return void
+     */
+    public function updateTraits(array $traits = [])
+    {
+        $existingTraits = [];
+        foreach ($this->identity_traits as $trait) {
+            $existingTraits[$trait->getTraitKey()] = $trait;
```

Comment:
Wow! I didn't know that Array indexes could be things other than integers in php. Interesting!